programs/cleos/main.cpp

```diff
@@ -3299,16 +3299,22 @@ int main( int argc, char** argv ) {
    string trx_json_to_sign;
    string str_private_key;
    string str_chain_id;
+   string str_private_key_file;
+   string str_keosd_pub_key;
    bool push_trx = false;
 
    auto sign = app.add_subcommand("sign", localized("Sign a transaction"), false);
    sign->add_option("transaction", trx_json_to_sign,
                                  localized("The JSON string or filename defining the transaction to sign"), true)->required();
    sign->add_option("-k,--private-key", str_private_key, localized("The private key that will be used to sign the transaction"));
+   sign->add_option("-w,--wallet", str_keosd_pub_key, localized("Ask keosd to sign with the given public key"));
```

Comment:
I think `--public-key` would be a better option name.
Use `key_store_executable_name` instead of `keosd`
The help text should say, ".. sign with the corresponding private key of the given public key" or something along those lines. It doesn't sign with the public key.

Reply:
Do we need option to sign via permission(s)? account@permission ?

Reply:
Make sense. PR updated with a better option name and description.

Reply:
I think signing via permission is more like a convenient option to add. But if we add it, I guess we will need to bring the auth_checker logic to cleos? Say, for example, alice@active with threshold 2 is satisfied by a key with weight 1 and bob@active with weight 1, and bob@weight is satisfied with a key. In this situation, if we decide to use `cleos sign` with `alice@active`, cleos will need to recursively decode the permission to end up with alice's key and bob's key and then sign with these two keys. These can already be achieved by passing public key of alice and bob to `cleos sign`'s `--public-key` option. Personally, I prefer to keep the logic in cleos simple and not bloated the software, so I'm not sure whether it's worth it or not to add this option.